src/language/language_type.rs

```diff
@@ -18,22 +18,26 @@ include!(concat!(env!("OUT_DIR"), "/language_type.rs"));
 impl LanguageType {
     /// Parses a given `Path` using the `LanguageType`. Returning `Stats`
     /// on success.
-    pub fn parse(self, path: PathBuf) -> io::Result<Stats> {
+    pub fn parse(self, path: PathBuf) -> Result<Stats, (io::Error, PathBuf)> {
```

Comment:
Instead of changing the API signature can we instead have a `parse_with_accuracy` or something similar that has this signature, and have `parse` call that and replace the `parse` call in `src/utils/fs.rs` with `parse_with_accuracy`?